### DISCUSSION TITLE
[Application Development] Napkin math gas estimation help needed

### DISCUSSION THREAD

**Discussion by TimDaub** [ethereum-optimism/developers#324] [Application Development] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hi, for some napkin math, I want to know what it would cost to store 29M in OP mainnet call data. I've found this page: https://docs.optimism.io/stack/transactions/fees. So, I'd use the ecotone formula? 

What I did:

- 28MB are 29360128 non-zero bytes
- OP base fee 0.05 gwei
- blob base fee 500 gwei

I transformed both of them into wei, and I used

```
tx_compressed_size = [(count_zero_bytes(tx_data)*4 + count_non_zero_bytes(tx_data)*16)] / 16
weighted_gas_price = 16*base_fee_scalar*base_fee + blob_base_fee_scalar*blob_base_fee
l1_data_fee = tx_compressed_size * weighted_gas_price
```

where I assumed that base_fee_scalar and blob_base_fee_scalar are 1. Where can I find out more about them?

Anyway, here is what I came up with:

29360128*16*50000000+500000000000 = 23488602400000000 wei or 0,0234886024 ETH. Does that make sense to you? It seems pretty cheap for 28MB

Frankly, with all these variables, that document is a bit above my head. Anyway, I just want to know if it's more in the order of 10000 EUR or 100 EUR, because if it's in the order of 100 EUR, then I think I want to continue building it.

**Comment by Chomtana** [comment]
A problem to consider is that 29,360,128 non-zero bytes would cost 29,360,128 * 16 = 469M gas, which exceeds the block limit of 30M gas. As a result, your application may not function as intended and might need to be submitted over multiple blocks. If your application operates 24/7, it could cause chain congestion, significantly increasing the OP base fee. This effect can be observed by looking at how dramatically the base fee is inflated.

**Reply by TimDaub** [reply]
ok fair but this would be a one time thing, so I could segment the data into a few separate txs and commit it. But other than that, do you see any glaring mistakes with my math?

**Reply by Chomtana** [reply, marked as answer]
You need to also consider the L2 gas part which is approximately 29360128 * 16 * 0.1 gwei ~ 46976204 gwei ~ 0.05 ETH in addition to the L1 fee.

Moreover, the base fee in the formula is not an OP base fee. It should be ETH mainnet base fee but since base_fee_scalar seems to be 0, that part is cancelled.

This cause `l1_data_fee = tx_compressed_size * blob_base_fee_scalar * blob_base_fee` that's pretty hard to accurately estimate but you can look into recent transactions on Etherscan and observe that usually `l1_data_fee < 4 * l2_fee`.

So, I think you are paying at most 0.05 + 4 * 0.05 = 0.25 ETH at most if you can split your data over a few thousand blocks.
